src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -106,6 +111,74 @@ public static Email parseEmail(String email) throws ParseException {
         return new Email(trimmedEmail);
     }
 
+    /**
+     * Parses a {@code String date} into a {@code Date}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code date} is invalid.
+     */
+    public static Date parseDate(String date) throws ParseException {
+        requireNonNull(date);
+        String trimmedDate = date.trim();
+        if (!Date.isValidDate(trimmedDate)) {
+            throw new ParseException(Date.MESSAGE_CONSTRAINTS);
+        }
+        return new Date(trimmedDate);
+    }
+
+    /**
+     * Parses a {@code String Time} into a {@code Time}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code Time} is invalid.
+     */
+    public static Time parseTime(String time) throws ParseException {
+        requireNonNull(time);
+        String trimmedTime = time.trim();
+        if (!Time.isValidTime(trimmedTime)) {
+            throw new ParseException(Time.MESSAGE_CONSTRAINTS);
+        }
+        return new Time(trimmedTime);
+    }
+
+    /**
+     * Parses 2 {@code String Time}s into {@code Time}s,
+     * then compare if  the range is valid. The first argument
+     * represents the start and the second, the end.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the time range is invalid.
+     */
+    public static void parseTimeRange(String startTime, String endTime) throws ParseException {
```

Comment:
I thought the function of the ParserUtil is to make sense of Strings or Collection<Strings>, which is why the parameters of _parseTimeRange_ are Strings, then I further parsed the parameters using _parseTime_, what do you think about this? I think your suggestion makes sense too but not sure if this deviates from the logic of ParserUtil since the documentation now states "Contains utility methods used for parsing strings in the various *Parser classes.

Reply:
oh ya makes sense... should pass in strings. But then it just does a validation check and doesn't return a result which is rather strange as well right.

Reply:
Hmm yeah, I thought of making it return a boolean but since it throws an exception if the TimeRange is invalid then I thought it wouldn't be necessary to return anything.